calculator_test.go

```diff
@@ -2,23 +2,416 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
+	"time"
 )
 
+// Note Fatal|FatalF are also useful methods,
+// to stop further execution within the same test.
+
+// Express test-case inputs and desired outputs
+type testCase struct {
+	a           float64
+	b           float64
+	want        float64
+	description string
+	errExpected bool
+}
+
+// This repeats the above struct,
+// but b is a slice-of-float64 instead.
+// Re-using the same struct for variadic and non-variadic would be nice,
+// but separate structs makes test more readable.
+type variadicTestCase struct {
+	a           float64
+	b           []float64
+	want        float64
+	description string
+	errExpected bool
+}
+
+// Test case for expressions (string parsed into numbers and an operator)
+type expressionTestCase struct {
+	e           string
+	want        float64
+	description string
+	errExpected bool
+}
+
+// Return a slice of N random numbers from 0-500
+func randFloat64Slice(n int) []float64 {
+	rand.Seed(time.Now().UnixNano())
+	// TODO: Should this be pre-allocated instead of using append()?
+	var r []float64
+
+	for i := 0; i < n; i++ {
+		x := rand.Float64() * float64(rand.Intn(500))
+		r = append(r, x)
+	}
+	return r
+}
+
 func TestAdd(t *testing.T) {
+	// Define test cases
+	//
+	// Reference RE: `cases := []testCases` vs. `var cases ...`
+	// ways to define this variable: https://blog.golang.org/slices-intro
+	testCases := []testCase{
+		{
+			description: "two positive numbers which sum to a positive",
+			a:           2,
+			b:           2,
+			want:        4,
+		},
+		{
+			description: "a positive and negative number which sum to a positive",
+			a:           7,
+			b:           -2,
+			want:        5,
+		},
+		{
+			description: "a positive and negative number which sum to a negative",
+			a:           3,
+			b:           -5,
+			want:        -2,
+		},
+	}
+
 	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+
+	// Fun fact: I initially called this variable `case` which is a keyword.
```

Comment:
Yes, you're not the only one to fall into that trap... or to use `t`, which doesn't work either!